c2rust-analyze/src/main.rs

```diff
@@ -150,6 +151,23 @@ fn run(tcx: TyCtxt) {
         gacx.fn_sigs.insert(ldid.to_def_id(), lsig);
     }
 
+    // Label the field types of each struct.
+    for defldid in tcx.hir_crate_items(()).definitions() {
+        let defdid = defldid.to_def_id();
+        if let DefKind::Struct = tcx.def_kind(defdid) {
+            let defty = tcx.type_of(defdid).kind();
+            if let TyKind::Adt(adtdef, _) = defty {
```

Comment:
You can get the `AdtDef` directly by calling `tcx.adt_def(defdid)`, instead of going through `type_of` + pattern matching.